components/teams-service/server/v2/server.go

```diff
@@ -62,6 +62,10 @@ func (s *Server) CreateTeam(ctx context.Context,
 	var team storage.Team
 	var err error
 	if team, err = s.service.Storage.StoreTeamWithProjects(ctx, req.Id, req.Name, req.Projects); err != nil {
+		// if the error is already a GRPC status code, return that directly.
+		if _, ok := status.FromError(err); ok {
+			return nil, err
+		}
```

Comment:
It's ok, but this does mean that a response from authz-service will look like it came from teams-service. I wonder if this could be confusing... 🤔

Reply:
I think we didn't want them to think about the proper response code, but I'm starting to think we should just return the two specific errors like we do internally in authz-service and let them return what they want per domain. We can think more on this.

Reply:
I like the idea of keeping "authz" decisions in authz as much as possible . . .will help with consistency. But whether that's confusing to the user is a different q